### PR TITLE
Add LastFM Support

### PR DESCRIPTION
**Pull request by hrfarmer**
Adds support for LastFM by using the LastFM API, which will let people link their LastFM account to the bot to be able to show whatever they're listening to in chat. Also I wasn't even able to test this again because something with osu broke that I was too lazy to fix, so let me know if it doesn't work on your end.

You will need to add `LASTFM_TOKEN` to your .env and get a token from [here](https://www.last.fm/api/account/create) (you'll need a LastFM account first)

Here's the database schema to add to your own database (I sure hope it's correct! As I said I wasn't able to really test anything since I couldn't get the bot running again)

``` sql
CREATE TABLE `lastfm` (
    -> `user_id` int unsigned NOT NULL DEFAULT '0',
    -> `lastfm_user` VARCHAR(36) DEFAULT NULL,
    -> PRIMARY KEY (`user_id`)
    -> ) ENGINE=InnoDB DEFAULT CHARSET=utf8mb4 COLLATE=utf8mb4_0900_ai_ci;
```

I completely messed up git while I tried to make this PR I think, so let me know if I horribly broke something.

### CODE REVIEW

File: main.py [hunk 1]
```diff
@@ -386,7 +389,6 @@ async def start(self):
 
     async def connect(self):
         await self.ws.send(f"PASS {self.oauth}")
-        print(f"> PASS {self.oauth}")
```
Comment (hrfarmer) [inline main.py:389]: oh i forgot to undo this change if you really need it to be there, i had just gotten rid of it while working on it cause i was working on it on stream and i do Not want the console to print my oauth

File: main.py [hunk 2]
```diff
@@ -2112,6 +2115,64 @@ async def antonyms_word(self, ctx):
...
+        # check whether they've linked before
+        if self.database.get_lastfm_user_from_user_id(ctx.user_id) is not None:
+            self.database.update_lastfm_data(ctx.user_id, username)
+        else:
+            self.database.new_lastfm_data(ctx.user_id, username)
+
+        await self.send_message(ctx.channel, f"@{ctx.user.display_name} Linked {username} to your account.")
+    
+    @command_manager.command("lastfm_username", aliases=["fmusername", "fmuser"])
+    async def lastfm_username(self, ctx):
```
Comment (Sheppsu) [inline main.py]: is there a use for this command

**Review by Sheppsu — CHANGES_REQUESTED** [review]

File: main.py (hunk 2 above)
Comment (hrfarmer) [reply main.py]: probably not honestly

**Review by Sheppsu — APPROVED** [review]
